### PR TITLE
Use matplotlib-inline instead of ipykernel.pylab

### PR DESCRIPTION
**Pull request by martinRenou**
Related to https://github.com/ipython/ipykernel/pull/591

### CODE REVIEW

File: IPython/core/display.py [hunk 1]
```diff
@@ -1173,16 +1173,8 @@ def set_matplotlib_formats(*formats, **kwargs):
...
-    from IPython.core.pylabtools import select_figure_formats
-    # build kwargs, starting with InlineBackend config
-    kw = {}
-    from ipykernel.pylab.config import InlineBackend
-    cfg = InlineBackend.instance()
-    kw.update(cfg.print_figure_kwargs)
-    kw.update(**kwargs)
-    shell = InteractiveShell.instance()
-    select_figure_formats(shell, formats, **kw)
+    from matplotlib_inline.backend_inline import set_matplotlib_formats_orig
```
Comment (martinRenou) [inline IPython/core/display.py:1176]: We might want to display a deprecated warning here

File: IPython/core/display.py [hunk 2]
```diff
@@ -1206,6 +1198,5 @@ def set_matplotlib_close(close=True):
         Should all matplotlib figures be automatically closed after each cell is
         run?
     """
-    from ipykernel.pylab.config import InlineBackend
-    cfg = InlineBackend.instance()
-    cfg.close_figures = close
+    from matplotlib_inline.backend_inline import set_matplotlib_close_orig
```
Comment (martinRenou) [inline IPython/core/display.py:1201]: and here

**Comment by Carreau** [comment]
@tacaswell would that also help or be the occasion to help cleaning some logic inside matplotlib ?

**Comment by martinRenou** [comment]
Tests are now passing and it seems to work as intended.

matplotlib

matplotlib-inline is installable from PyPi or conda-forge. The repo is here for now: https://github.com/martinRenou/matplotlib-inline. But I guess we should put this repo under the Matplotlib or the ipython organization.

**Comment by martinRenou** [comment]
It is indeed one good step for cleaning the Matplotlib logic in IPython

**Comment by Carreau** [comment]
I've push minor changes (mostly saying it's deprecated since 7.23), and will merge once it passes.

**Comment by martinRenou** [comment]
Thanks!

**Comment by SylvainCorlay** [comment]
Hey, should we go ahead with this and move `matplotlib_inline` to this organization?

**Comment by Carreau** [comment]
> Hey, should we go ahead with this and move `matplotlib_inline` to this organization?

Up to you; I know there are some effort to cleanup what is where; but I'll adapt if you wish to keep it somewhere else. Whatever suit people.

**Comment by Carreau** [comment]
Also forgot to merge, which is now done.

**Comment by martinRenou** [comment]
Thanks for merging! 

I guess we can move the `matplotlib-inline` repo under the IPython org then? The inline backend is quite related to IPython.

**Comment by tacaswell** [comment]
Sorry for late commits.  I assume this does not make IPython have an install-time dependency on Matplotilb?

**Comment by martinRenou** [comment]
Yes, I didn't make `matplotlib` a `matplotlib-inline` dependency :)

**Comment by SylvainCorlay** [comment]
Hey @Carreau, how about we tag a pre-release of IPython with that change?
We could start testing the companion PR in ipykernel...

**Pull request merged**